tests/models/test_modeling_common.py

```diff
@@ -732,3 +733,26 @@ def test_push_to_hub_in_organization(self):
 
         # Reset repo
         delete_repo(self.org_repo_id, token=TOKEN)
+
+    @unittest.skipIf(
+        not is_jinja_available(),
+        reason="Model card tests cannot be performed with Jinja installed.",
+    )
+    def test_push_to_hub_library_name(self):
```

Comment:
(nit) I would add an explicit test both for when the model card doesn't exist yet and for when the model card already exists. Maybe not needed to test the full `push_to_hub` method but simply the `create_and_tag_model_card` helper (or whatever its name :) )

Reply:
Does 99ce47c8 work for you?

Reply:
Oh,  I didn't notice that there is a `generate_model_card` and a `create_model_card` [in the hub utils](https://github.com/huggingface/diffusers/blob/19d26da8bc9d38bc5e2d3d95b90111735f84fc32/src/diffusers/utils/hub_utils.py#L98). Should we merge them since they seem to do closely related things?  (the difference is generating for a training or generating from anywhere, right?). Naming is misleading in that case (sorry, didn't notice before when I suggest `generate_model_card`).

Regarding the test, yes it looks good to me :)

Reply:
Sorry, but it won't work since `tmpdir` is local. What's the best way to test here?

Reply:
@Wauplin I merged them. However, I am not sure about the test since `tmpdir` is local and `ModelCard.load()` would fail there.

Reply:
I think tests are good. I've added a comment below to test from existing file with existing library_name that is not diffusers.

But what I meant above is that with this PR the hub utils feels clunky. We now have:
- `create_model_card` that creates a model card for a training. The method looks outdated and not used anywhere. However it introduces a template that is nice.
- `generate_model_card` that either loads an existing model card or create a new one (from a different template) and add `library_name: diffusers` to it. This method is used in the codebase.

---

Maybe what I would do to solve this (and sorry if it's a revamp of the PR):
- deprecate `create_model_card` (or even remove it completely) if it's not used
- add in `hub_utils.py` a `load_or_create_model_card` helper that returns a `ModelCard` object without modifying anything. It's similar to the `try: (ModelCard.load) except EntryNotFound: (ModelCard.from_template)` part
- add in `hub_utils.py` a `populate_model_card` that takes as input a `ModelCard` object and add `library_name: diffusers` if doesn't exist yet.
- then in the codebase (for example in `pipeline_utils.py`), you would do
```py
model_card = load_or_create_model_card(repo_id, token=token, is_pipeline=True)
populate_model_card(model_card)
model_card.save(os.path.join(save_directory, "README.md"))
```

WDTY? 

(to take with a grain of salt, I'm not expert in `diffusers` codebase so I might be missing some parts)

Reply:
(sorry @sayakpaul I didn't see your comment while posting this message)

Reply:
Makes sense. The latest commit should be have reflected these. Let me know if that makes sense. 

I have opted to remove `create_model_card()` and the related test as it's not really used.